mongomock/collection.py

```diff
@@ -1649,6 +1649,30 @@ def _extend_collection(out_collection, field, expression):
                 if k == '$match':
                     out_collection = [doc for doc in out_collection
                                       if filter_applies(v, doc)]
+                elif k == '$lookup':
+                    for operator in ['let', 'pipeline']:
+                        if operator in stage['$lookup']:
+                            raise NotImplementedError(
+                                "Although %s is a valid lookup operator for the "
+                                "aggregation pipeline, it is currently not "
+                                "implemented in Mongomock." % operator)
+                    for operator in ['from', 'localField', 'foreignField', 'as']:
+                        if operator not in stage['$lookup']:
+                            raise OperationFailure(
+                                "must specify '%s' field for a $lookup" % operator)
+                        if not isinstance(stage['$lookup'][operator], str):
```

Comment:
You should also raise an error `pymongo.errors.OperationFailure: FieldPath field names may not start with '$'.` if localField or foreignField starts with a `$`.